src/ansys/fluent/core/solver/function/reduction.py

```diff
@@ -166,18 +174,30 @@ def _eval_reduction(solver, reduction, locations, expr=None):
     )
 
 
-def _extent_expression(f_string, extent_name, expr, locations, ctxt):
+def _extent_expression(
+    f_string, extent_name, expr, locations, ctxt, weight=None, condition=None
+):
     locns = _locns(locations, ctxt)
     numerator = 0.0
     denominator = 0.0
     for solver, names in locns:
         solver = solver or _root(ctxt)
-        val = _eval_reduction(solver, f_string, names, expr)
-        extent = _eval_reduction(solver, extent_name, names) if len(locns) > 1 else 1
+        val = _eval_reduction(
+            solver, f_string, names, expr, weight=weight, condition=condition
+        )
+        extent = (
+            _eval_reduction(
+                solver, extent_name, names, weight=weight, condition=condition
+            )
+            if len(locns) > 1
+            else 1
+        )
         try:
             numerator += val * extent
             denominator += extent
         except TypeError:
+            if type(val) == list:
+                return val
```

Comment:
Should we not need to compute the weighted average in this case?

Reply:
@mkundu1, Done, Thank you.